versions.tf

```
@@ -9,7 +9,7 @@ terraform {
   required_providers {
     github = {
       source  = "integrations/github"
-      version = ">= 4.20, < 6.0"
+      version = ">= 5.8, < 6.0"
```

Since the `allow_update_branch` argument was added in [v5.7.0](https://github.com/integrations/terraform-provider-github/releases/tag/v5.7.0). It would be better to keep the minimum version at v5.7.0.

```suggestion
      version = ">= 5.7, < 6.0"
```

---

I understand what you mean, but #156 requires also a newer version (even higher) so therefore I don't see a reason to not bump it to the same level.

---

I agree, but both changes are independent. They could be merged independently at different points in time depending on the maintainers' discretion of release cycles.

If only this pull request were to be merged without merging #156 along with it, then we would be not be bumping to the most minimum version required for the module to work as expected. And it would be confusing for future commit inspectors to understand the rationale for making the bump.

This is only my personal opinion. I am not a maintainer for this module, so I am not speaking with that authority. So, it is at your discretion to make changes if you prefer.

I very much appreciate the time and effort you put in raising this pull request.